### DISCUSSION TITLE
Add a custom field into medusa inventory (inventory_level) but not returned as part of the response fields

### DISCUSSION THREAD

**Discussion by abidnasratyar01** [medusajs/medusa#5499] [Q&A]
src\models\inventory_level.ts 
import { InventoryLevel as MedusaInventoryLevel} from "@medusajs/inventory/dist/models/inventory-level";
import { Column, Entity } from "typeorm";

@Entity()
export class InventoryLevel extends MedusaInventoryLevel {
    @Column({ type: "int", nullable: true })
    custom: number;
}

src\index.d.ts
export declare module "@medusajs/inventory/dist/models/inventory-level" {
	declare interface InventoryLevel {
		custom?: number 
	}
}

src\loaders\extend-inventory-level-fields.ts
export default async function () {
    const imports = (await import(
      "@medusajs/medusa/dist/api/routes/admin/inventory-items/index"
    )) as any
    imports.defaultAdminLocationLevelFields = [
      ...imports.defaultAdminLocationLevelFields,
      "custom",
    ]
  }
  
  
  Field added to DB as well
  ALTER TABLE INVENTORY_LEVEL ADD custom integer NULL
  
  
  
 As mentioned in title I am not able to get this new custom field in response body for Admin UI inventory.. Would appreciate any help in this regard.

**Comment by srindom** [comment]
Could you add some more context about what data the custom field holds? Maybe there's a better approach to take here.

**Reply by abidnasratyar01** [reply]
mr srindom, 

I want to add a field to inventory level, to hold the quantity of on the way/ incoming inventory items. Inventory level do have a field incoming_quantity, which isn't used for any purpose right now but down the road it could be used for some other specific purpose hence could collide with my specific purpose and create ambiguity. Therefore, I do not prefer to use incoming_quantity and wanted to add a new field.

**Reply by abidnasratyar01** [reply]
what do you think?
@adrien2p @olivermrbl @fPolic
